### PR TITLE
Basgra updates including the YASSO SOC model

### PR DESCRIPTION
**Pull request by juliusvira**
## Description
This PR adds a new soil organic carbon (SOC) model to the BASGRA grassland model, updates the wrappers, and adds unit tests for the wrappers. The SOC model is an adaptation of the YASSO20 (https://doi.org/10.5194/gmd-15-1735-2022); the user can switch between YASSO20 and the original BASGRA_N SOC model by setting a parameter. The YASSO implementation also includes an experimental nitrogen cycling extension, which can be optionally enabled. 

## Motivation and Context
The modifications to BASGRA provide an option to choose a more detailed SOC model when needed. 

## Review Time Estimate

- [ ] Immediately
- [ ] Within one week
- [x] When possible

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue) 
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] My name is in the list of CITATION.cff
- [ ] I have updated the CHANGELOG.md.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: models/basgra/R/write.config.BASGRA.R [hunk 1]
```diff
@@ -372,7 +594,7 @@ write.config.BASGRA <- function(defaults, trait.values, settings, run.id, IC = N
     }
     
     # Initial mineral N
-    nmin0 <- try(ncdf4::ncvar_get(IC.nc, "soil_nitrogen_content"), silent = TRUE)
+    nmin0 <- try(ncdf4::ncvar_get(IC.nc, "soil_inorganic_nitrogen_content"), silent = TRUE)
     if (!is.na(nmin0) && is.numeric(nmin0)) {
       run_params[which(names(run_params) == "NMIN0")] <- PEcAn.utils::ud_convert(nmin0, "kg", "g")
     }
```
Comment (istfer) [inline models/basgra/R/write.config.BASGRA.R:600]:
```suggestion
    }
    
        # Initial organic N
    nsom0 <- try(ncdf4::ncvar_get(IC.nc, "soil_organic_nitrogen_content"), silent = TRUE)
    if (!is.na(nsom0) && is.numeric(nsom0)) {
      run_params[which(names(run_params) == "NSOM")] <- PEcAn.utils::ud_convert(nsom0, "kg", "g")
    }
```
maybe let's add this too

**Review by istfer — APPROVED** [review]

File: models/basgra/R/write.config.BASGRA.R [hunk 2]
```diff
@@ -248,79 +275,274 @@ write.config.BASGRA <- function(defaults, trait.values, settings, run.id, IC = N
...
+      last_somf <- sum(last_vals[names(last_vals) == "CSOM_A"],
+                       last_vals[names(last_vals) == "CSOM_W"],
+                       last_vals[names(last_vals) == "CSOM_E"],
+                       last_vals[names(last_vals) == "CSOM_N"])
+      
+      last_soms <- last_vals[names(last_vals) == "CSOM_H"]
+      
+      if ("fast_soil_pool_carbon_content"  %in% ic.names & "slow_soil_pool_carbon_content"  %in% ic.names) {
+        
+        new_somf <- udunits2::ud.convert(IC$fast_soil_pool_carbon_content, "kg", "g")
```
Comment (istfer) [inline models/basgra/R/write.config.BASGRA.R:330]:
```suggestion
        new_somf <- PEcAn.utils::ud_convert(IC$fast_soil_pool_carbon_content, "kg", "g") 
```

File: models/basgra/R/write.config.BASGRA.R [hunk 3]
```diff
@@ -248,79 +275,274 @@ write.config.BASGRA <- function(defaults, trait.values, settings, run.id, IC = N
...
+                       last_vals[names(last_vals) == "CSOM_W"],
+                       last_vals[names(last_vals) == "CSOM_E"],
+                       last_vals[names(last_vals) == "CSOM_N"])
+      
+      last_soms <- last_vals[names(last_vals) == "CSOM_H"]
+      
+      if ("fast_soil_pool_carbon_content"  %in% ic.names & "slow_soil_pool_carbon_content"  %in% ic.names) {
+        
+        new_somf <- udunits2::ud.convert(IC$fast_soil_pool_carbon_content, "kg", "g") 
+        new_soms <- udunits2::ud.convert(IC$slow_soil_pool_carbon_content, "kg", "g")
```
Comment (istfer) [inline models/basgra/R/write.config.BASGRA.R:331]:
```suggestion
        new_soms <- PEcAn.utils::ud_convert(IC$slow_soil_pool_carbon_content, "kg", "g") 
```

File: models/basgra/R/write.config.BASGRA.R [hunk 4]
```diff
@@ -248,79 +275,274 @@ write.config.BASGRA <- function(defaults, trait.values, settings, run.id, IC = N
...
+        run_params[names(run_params) == "CSOM_E"] <- last_vals[names(last_vals) == "CSOM_E"]
+        run_params[names(run_params) == "CSOM_N"] <- last_vals[names(last_vals) == "CSOM_N"]
+        run_params[names(run_params) == "CSOM_H"] <- last_soms
+        
+        run_params[names(run_params) == "NSOM"] <- last_vals[names(last_vals) == "NSOM"]
+        
+      }
+      
+      # #else-if ("TotSoilCarb"  %in% ic.names)?
+      # new_totc <- udunits2::ud.convert(IC$TotSoilCarb, "kg", "g")
```
Comment (istfer) [inline models/basgra/R/write.config.BASGRA.R:357]:
```suggestion
      # new_totc <- PEcAn.utils::ud_convert(IC$TotSoilCarb, "kg", "g") 
```

File: models/basgra/R/write.config.BASGRA.R [hunk 5]
```diff
@@ -248,79 +275,274 @@ write.config.BASGRA <- function(defaults, trait.values, settings, run.id, IC = N
...
+      # run_params[names(run_params) == "CSOM_A"] <- ratio_soc * last_vals[names(last_vals) == "CSOM_A"]
+      # run_params[names(run_params) == "CSOM_W"] <- ratio_soc * last_vals[names(last_vals) == "CSOM_W"]
+      # run_params[names(run_params) == "CSOM_E"] <- ratio_soc * last_vals[names(last_vals) == "CSOM_E"]
+      # run_params[names(run_params) == "CSOM_N"] <- ratio_soc * last_vals[names(last_vals) == "CSOM_N"]
+      # run_params[names(run_params) == "CSOM_H"] <- ratio_soc * last_vals[names(last_vals) == "CSOM_H"]
+      
+      
+    }else{ # no Yasso
+      if ("fast_soil_pool_carbon_content"  %in% ic.names) {
+        run_params[names(run_params) == "CSOMF0"] <- udunits2::ud.convert(IC$fast_soil_pool_carbon_content, "kg", "g")
```
Comment (istfer) [inline models/basgra/R/write.config.BASGRA.R:371]:
```suggestion
        run_params[names(run_params) == "CSOMF0"] <- PEcAn.utils::ud_convert(IC$fast_soil_pool_carbon_content, "kg", "g")
```

File: models/basgra/R/write.config.BASGRA.R [hunk 6]
```diff
@@ -248,79 +275,274 @@ write.config.BASGRA <- function(defaults, trait.values, settings, run.id, IC = N
...
+      }else{
+        run_params[names(run_params) == "CSOMF0"] <- last_vals[names(last_vals) == "CSOMF"]
+      }
+      run_params[names(run_params) == "NSOMF0"]  <- run_params[names(run_params) == "CSOMF0"] / run_params[names(run_params) == "CNSOMF0"]
+      
+      if ("slow_soil_pool_carbon_content"  %in% ic.names) {
+        run_params[names(run_params) == "CSOMS0"] <- udunits2::ud.convert(IC$slow_soil_pool_carbon_content, "kg", "g")
```
Comment (istfer) [inline models/basgra/R/write.config.BASGRA.R:378]:
```suggestion
        run_params[names(run_params) == "CSOMS0"] <- PEcAn.utils::ud_convert(IC$slow_soil_pool_carbon_content, "kg", "g")
```

File: models/basgra/R/write.config.BASGRA.R [hunk 7]
```diff
@@ -248,79 +275,274 @@ write.config.BASGRA <- function(defaults, trait.values, settings, run.id, IC = N
...
+      }else{
+        run_params[names(run_params) == "CSOMS0"] <- last_vals[names(last_vals) == "CSOMS"]
+      }
+      run_params[names(run_params) == "NSOMS0"]  <- run_params[names(run_params) == "CSOMS0"] / run_params[names(run_params) == "CNSOMS0"]
+      
     }
     
     if ("CropYield"  %in% ic.names) {
-       run_params[names(run_params) == "YIELDI"] <-  PEcAn.utils::ud_convert(IC$CropYield, "kg", "g")
+      run_params[names(run_params) == "YIELDI"] <- udunits2::ud.convert(IC$CropYield, "kg", "g")
```
Comment (istfer) [inline models/basgra/R/write.config.BASGRA.R:387]:
```suggestion
      run_params[names(run_params) == "YIELDI"] <- PEcAn.utils::ud_convert(IC$CropYield, "kg", "g")
```

File: models/basgra/R/write.config.BASGRA.R [hunk 8]
```diff
@@ -248,79 +275,274 @@ write.config.BASGRA <- function(defaults, trait.values, settings, run.id, IC = N
...
+    }else{
+      run_params[names(run_params) == "YIELDI"]  <- last_vals[names(last_vals) == "YIELD_POT"]
     }
     
     if ("litter_carbon_content"  %in% ic.names) {
-      run_params[names(run_params) == "CLITT0"] <-  PEcAn.utils::ud_convert(IC$litter_carbon_content, "kg", "g")
+      run_params[names(run_params) == "CLITT0"] <- udunits2::ud.convert(IC$litter_carbon_content, "kg", "g")
```
Comment (istfer) [inline models/basgra/R/write.config.BASGRA.R:393]:
```suggestion
      run_params[names(run_params) == "CLITT0"] <- PEcAn.utils::ud_convert(IC$litter_carbon_content, "kg", "g")
```

File: models/basgra/R/write.config.BASGRA.R [hunk 9]
```diff
@@ -248,79 +275,274 @@ write.config.BASGRA <- function(defaults, trait.values, settings, run.id, IC = N
...
     }
+    #run_params[names(run_params) == "NLITT0"] <- run_params[names(run_params) == "CLITT0"] / run_params[names(run_params) == "CNLITT0"]
+    run_params[which(names(run_params) == "NLITT0")] <- last_vals[names(last_vals) == "NLITT"]
     
-    # not as important as others but you can throw this into the SDA too, then comment out last value overwriting below
-    # if ("stubble_carbon_content"  %in% ic.names) {
-    #   run_params[names(run_params) == "CSTUBI"] <-  PEcAn.utils::ud_convert(IC$stubble_carbon_content, "kg", "g")
-    # }
+    if ("stubble_carbon_content"  %in% ic.names) {
+      run_params[names(run_params) == "CSTUBI"] <- udunits2::ud.convert(IC$stubble_carbon_content, "kg", "g")
```
Comment (istfer) [inline models/basgra/R/write.config.BASGRA.R:401]:
```suggestion
      run_params[names(run_params) == "CSTUBI"] <- PEcAn.utils::ud_convert(IC$stubble_carbon_content, "kg", "g")
```

File: models/basgra/R/write.config.BASGRA.R [hunk 10]
```diff
@@ -248,79 +275,274 @@ write.config.BASGRA <- function(defaults, trait.values, settings, run.id, IC = N
...
+    }else{
+      run_params[names(run_params) == "CSTUBI"] <- last_vals[names(last_vals) == "CSTUB"]
+    }
    
     if ("stem_carbon_content"  %in% ic.names) {
-      run_params[names(run_params) == "CSTI"] <-  PEcAn.utils::ud_convert(IC$stem_carbon_content, "kg", "g")
+      run_params[names(run_params) == "CSTI"] <- udunits2::ud.convert(IC$stem_carbon_content, "kg", "g")
```
Comment (istfer) [inline models/basgra/R/write.config.BASGRA.R:407]:
```suggestion
      run_params[names(run_params) == "CSTI"] <- PEcAn.utils::ud_convert(IC$stem_carbon_content, "kg", "g")
```

File: models/basgra/R/write.config.BASGRA.R [hunk 11]
```diff
@@ -248,79 +275,274 @@ write.config.BASGRA <- function(defaults, trait.values, settings, run.id, IC = N
...
+    }else{
+      run_params[names(run_params) == "CSTI"] <- last_vals[names(last_vals) == "CST"]
     }
     
+    # NRT        = NCR * CRTI
+    #run_params[names(run_params) == "NCR"] <- last_vals[names(last_vals) == "NRT"] / last_vals[names(last_vals) == "CRT"] 
     if ("root_carbon_content"  %in% ic.names) {
-      run_params[names(run_params) == "LOG10CRTI"] <-  PEcAn.utils::ud_convert(IC$root_carbon_content, "kg", "g")
+      run_params[names(run_params) == "LOG10CRTI"] <-  udunits2::ud.convert(IC$root_carbon_content, "kg", "g")
```
Comment (istfer) [inline models/basgra/R/write.config.BASGRA.R:415]:
```suggestion
      run_params[names(run_params) == "LOG10CRTI"] <-  PEcAn.utils::ud_convert(IC$root_carbon_content, "kg", "g")
```

File: models/basgra/R/write.config.BASGRA.R [hunk 12]
```diff
@@ -248,79 +275,274 @@ write.config.BASGRA <- function(defaults, trait.values, settings, run.id, IC = N
...
+    lai_tmp <- run_params[names(run_params) == "LOG10LAII"]
+    ncshi <- run_params[names(run_params) == "NCSHMAX"] *
+      (1-exp(-run_params[names(run_params) == "K"]*lai_tmp)) / (run_params[names(run_params) == "K"]*lai_tmp)
+    run_params[which(names(run_params) == "NSHI")] <- ncshi *
+      ((run_params[names(run_params) == "LOG10CLVI"]) + run_params[names(run_params) == "CSTI"])
+    
 
     if ("reserve_carbon_content"  %in% ic.names) {
-      run_params[names(run_params) == "LOG10CRESI"] <-  PEcAn.utils::ud_convert(IC$reserve_carbon_content, "kg", "g")
+      run_params[names(run_params) == "LOG10CRESI"] <- udunits2::ud.convert(IC$reserve_carbon_content, "kg", "g")
```
Comment (istfer) [inline models/basgra/R/write.config.BASGRA.R:432]:
```suggestion
      run_params[names(run_params) == "LOG10CRESI"] <- PEcAn.utils::ud_convert(IC$reserve_carbon_content, "kg", "g")
```

File: models/basgra/R/write.config.BASGRA.R [hunk 13]
```diff
@@ -248,79 +275,274 @@ write.config.BASGRA <- function(defaults, trait.values, settings, run.id, IC = N
...
+    }else{
+      run_params[names(run_params) == "LOG10CRESI"] <- last_vals[names(last_vals) == "CRES"]
     }
     
     if ("leaf_carbon_content"  %in% ic.names) {
-      run_params[names(run_params) == "LOG10CLVI"] <-  PEcAn.utils::ud_convert(IC$leaf_carbon_content, "kg", "g")
+      run_params[names(run_params) == "LOG10CLVI"] <- udunits2::ud.convert(IC$leaf_carbon_content, "kg", "g")
```
Comment (istfer) [inline models/basgra/R/write.config.BASGRA.R:438]:
```suggestion
      run_params[names(run_params) == "LOG10CLVI"] <- PEcAn.utils::ud_convert(IC$leaf_carbon_content, "kg", "g")
```

File: models/basgra/R/write.config.BASGRA.R [hunk 14]
```diff
@@ -248,79 +275,274 @@ write.config.BASGRA <- function(defaults, trait.values, settings, run.id, IC = N
...
+    }else{
+      run_params[names(run_params) == "LOG10CLVI"] <- last_vals[names(last_vals) == "CLV"]
     }
     
     if ("dead_leaf_carbon_content"  %in% ic.names) {
-      run_params[names(run_params) == "CLVDI"] <-  PEcAn.utils::ud_convert(IC$dead_leaf_carbon_content, "kg", "g")
+      run_params[names(run_params) == "CLVDI"] <- udunits2::ud.convert(IC$dead_leaf_carbon_content, "kg", "g")
```
Comment (istfer) [inline models/basgra/R/write.config.BASGRA.R:444]:
```suggestion
      run_params[names(run_params) == "CLVDI"] <- PEcAn.utils::ud_convert(IC$dead_leaf_carbon_content, "kg", "g")
```

File: models/basgra/R/run_BASGRA.R [hunk 15]
```diff
@@ -426,6 +489,29 @@ run_BASGRA <- function(run_met, run_params, site_harvest, site_fertilize, start_
...
+    # WCL = WAL*0.001 / (ROOTD-Fdepth) Water concentration in non-frozen soil
+    # need to think about ice! but the sensors maybe don't measure that
+    ROOTD  <- output[thisyear, which(outputNames == "ROOTD")] 
+    Fdepth <- output[thisyear, which(outputNames == "Fdepth")] 
+    outlist[[length(outlist)+1]]  <- soilm * 0.001 / (ROOTD - Fdepth)
+    
+    # Additional C fluxes
+    outlist[[length(outlist)+1]] <- udunits2::ud.convert(output[thisyear, outputNames == "FLITTC_LEAF"],
+                                                         "g m-2", "kg m-2") / sec_in_day    
+    outlist[[length(outlist)+1]] <- udunits2::ud.convert(output[thisyear, outputNames == "FLITTC_ROOT"],
```
Comment (istfer) [inline models/basgra/R/run_BASGRA.R:502]:
```suggestion
    outlist[[length(outlist)+1]] <- use PEcAn.utils::ud_convert(output[thisyear, outputNames == "FLITTC_ROOT"],
```

File: models/basgra/R/run_BASGRA.R [hunk 16]
```diff
@@ -426,6 +489,29 @@ run_BASGRA <- function(run_met, run_params, site_harvest, site_fertilize, start_
...
+                                                         "g m-2", "kg m-2") / sec_in_day
+    outlist[[length(outlist)+1]] <- udunits2::ud.convert(output[thisyear, outputNames == "FHARVC"],
```
Comment (istfer) [inline models/basgra/R/run_BASGRA.R:504]:
```suggestion
    outlist[[length(outlist)+1]] <- use PEcAn.utils::ud_convert(output[thisyear, outputNames == "FHARVC"],
```

File: models/basgra/R/run_BASGRA.R [hunk 17]
```diff
@@ -426,6 +489,29 @@ run_BASGRA <- function(run_met, run_params, site_harvest, site_fertilize, start_
...
+                                                         "g m-2", "kg m-2") / sec_in_day
+    outlist[[length(outlist)+1]] <- udunits2::ud.convert(output[thisyear, outputNames == "NEE"],
```
Comment (istfer) [inline models/basgra/R/run_BASGRA.R:506]:
```suggestion
    outlist[[length(outlist)+1]] <- use PEcAn.utils::ud_convert(output[thisyear, outputNames == "NEE"],
```

File: models/basgra/R/run_BASGRA.R [hunk 18]
```diff
@@ -426,6 +489,29 @@ run_BASGRA <- function(run_met, run_params, site_harvest, site_fertilize, start_
...
+                                                         "g m-2", "kg m-2") / sec_in_day
+    outlist[[length(outlist)+1]] <- udunits2::ud.convert(output[thisyear, outputNames == "FRUNOFFC"],
+                                                         "g m-2", "kg m-2") / sec_in_day
+    outlist[[length(outlist)+1]] <- udunits2::ud.convert(output[thisyear, outputNames == "FSOILAMDC"],
```
Comment (istfer) [inline models/basgra/R/run_BASGRA.R:510]:
```suggestion
    outlist[[length(outlist)+1]] <- use PEcAn.utils::ud_convert(output[thisyear, outputNames == "FSOILAMDC"],
```

File: models/basgra/R/run_BASGRA.R [hunk 19]
```diff
@@ -310,37 +328,66 @@ run_BASGRA <- function(run_met, run_params, site_harvest, site_fertilize, start_
   calendar_Ndep[2,] <- c( 1980, 366,  0*1000/(10000*365) ) #  0 kg N ha-1 y-1 N-deposition in 1980
   calendar_Ndep[3,] <- c( 2100, 366,  0*1000/(10000*365) ) #  0 kg N ha-1 y-1 N-deposition in 2100
   
-  days_harvest      <- matrix(as.integer(-1), nrow= 300, ncol = 3)
+  harvest_params <- matrix(0.0, nrow=300, ncol=2)
+  df_harvest <- read.csv(site_harvest)
```
Comment (istfer) [inline models/basgra/R/run_BASGRA.R:332]:
```suggestion
  df_harvest <- utils::read.csv(site_harvest)
```

File: models/basgra/R/run_BASGRA.R [hunk 20]
```diff
@@ -310,37 +328,66 @@ run_BASGRA <- function(run_met, run_params, site_harvest, site_fertilize, start_
...
                      NDAYS,
                      NOUT,
-                     matrix(0, NDAYS, NOUT))[[8]]
-  
+                     matrix(0, NDAYS, NOUT))[[10]]
   # for now a hack to write other states out
-  save(output, file = file.path(outdir, "output_basgra.Rdata"))
+  # save(output, file = file.path(outdir, "output_basgra.Rdata"))
+  if (write_raw_output) {
+    write.csv(setNames(as.data.frame(output), outputNames), file.path(outdir, "output_basgra.csv"))
```
Comment (istfer) [inline models/basgra/R/run_BASGRA.R:389]:
```suggestion
    utils::write.csv(setNames(as.data.frame(output), outputNames), file.path(outdir, "output_basgra.csv"))
```

File: models/basgra/R/run_BASGRA.R [hunk 21]
```diff
@@ -426,6 +489,29 @@ run_BASGRA <- function(run_met, run_params, site_harvest, site_fertilize, start_
...
     outlist[[length(outlist)+1]]  <- PEcAn.utils::ud_convert(soilm, "mm", "m") * 1000 # (kg m-3) density of water in soil
+
+    # WCL = WAL*0.001 / (ROOTD-Fdepth) Water concentration in non-frozen soil
+    # need to think about ice! but the sensors maybe don't measure that
+    ROOTD  <- output[thisyear, which(outputNames == "ROOTD")] 
+    Fdepth <- output[thisyear, which(outputNames == "Fdepth")] 
+    outlist[[length(outlist)+1]]  <- soilm * 0.001 / (ROOTD - Fdepth)
+    
+    # Additional C fluxes
+    outlist[[length(outlist)+1]] <- use PEcAn.utils::ud_convert(output[thisyear, outputNames == "FLITTC_LEAF"],
```
Comment (istfer) [inline models/basgra/R/run_BASGRA.R:500]:
```suggestion
    outlist[[length(outlist)+1]] <- PEcAn.utils::ud_convert(output[thisyear, outputNames == "FLITTC_LEAF"],
```
sorry that this crept in

File: models/basgra/R/run_BASGRA.R [hunk 22]
```diff
@@ -426,6 +489,29 @@ run_BASGRA <- function(run_met, run_params, site_harvest, site_fertilize, start_
...
+                                                         "g m-2", "kg m-2") / sec_in_day    
+    outlist[[length(outlist)+1]] <- use PEcAn.utils::ud_convert(output[thisyear, outputNames == "FLITTC_ROOT"],
```
Comment (istfer) [inline models/basgra/R/run_BASGRA.R:502]:
```suggestion
    outlist[[length(outlist)+1]] <- PEcAn.utils::ud_convert(output[thisyear, outputNames == "FLITTC_ROOT"],
```

File: models/basgra/R/run_BASGRA.R [hunk 23]
```diff
@@ -426,6 +489,29 @@ run_BASGRA <- function(run_met, run_params, site_harvest, site_fertilize, start_
...
+                                                         "g m-2", "kg m-2") / sec_in_day
+    outlist[[length(outlist)+1]] <- use PEcAn.utils::ud_convert(output[thisyear, outputNames == "FHARVC"],
```
Comment (istfer) [inline models/basgra/R/run_BASGRA.R:504]:
```suggestion
    outlist[[length(outlist)+1]] <- PEcAn.utils::ud_convert(output[thisyear, outputNames == "FHARVC"],
```

File: models/basgra/R/run_BASGRA.R [hunk 24]
```diff
@@ -426,6 +489,29 @@ run_BASGRA <- function(run_met, run_params, site_harvest, site_fertilize, start_
...
+                                                         "g m-2", "kg m-2") / sec_in_day
+    outlist[[length(outlist)+1]] <- use PEcAn.utils::ud_convert(output[thisyear, outputNames == "NEE"],
+                                                         "g m-2", "kg m-2") / sec_in_day
+    outlist[[length(outlist)+1]] <- use PEcAn.utils::ud_convert(output[thisyear, outputNames == "FRUNOFFC"],
```
Comment (istfer) [inline models/basgra/R/run_BASGRA.R:508]:
```suggestion
    outlist[[length(outlist)+1]] <- PEcAn.utils::ud_convert(output[thisyear, outputNames == "FRUNOFFC"],
```

File: models/basgra/R/write.config.BASGRA.R [hunk 25]
```diff
@@ -31,10 +31,13 @@ write.config.BASGRA <- function(defaults, trait.values, settings, run.id, IC = N
   outdir <- file.path(settings$host$outdir, run.id)
   
   # load default(!) BASGRA params
-  run_params <- PEcAn.utils::load_local(system.file("BASGRA_params.Rdata", package = "PEcAn.BASGRA"))$default_params
-  
+  if (!is.null(settings$run$inputs$defaults$path)) {
+    df_run_params <- read.csv(settings$run$inputs$defaults$path)
```
Comment (istfer) [inline models/basgra/R/write.config.BASGRA.R:35]:
```suggestion
    df_run_params <- utils::read.csv(settings$run$inputs$defaults$path)
```

File: models/basgra/R/write.config.BASGRA.R [hunk 26]
```diff
@@ -31,10 +31,13 @@ write.config.BASGRA <- function(defaults, trait.values, settings, run.id, IC = N
...
+  } else {
+    df_run_params <- read.csv(system.file("BASGRA_params.csv", package = "PEcAn.BASGRA"))
```
Comment (istfer) [inline models/basgra/R/write.config.BASGRA.R:37]:
```suggestion
    df_run_params <- utils::read.csv(system.file("BASGRA_params.csv", package = "PEcAn.BASGRA"))
```

**Comment by infotroph** [comment]
Sorry for the long wait on this. @istfer @juliusvira is this ready to merge as soon as tests pass?

File: models/basgra/tests/testthat/ic_with_yasso_pools.nc
Comment (infotroph) [inline models/basgra/tests/testthat/ic_with_yasso_pools.nc:1]: I started a comment about how these 12-MB IC files were too big for an R package, but on testing they compress remarkably well -- the entire PEcAn.BASGRA_1.7.2.tar.gz after this PR is still only 118 KB! So I think there's no action needed.

**Comment by mdietze** [comment]
@juliusvira @istfer would love to get this PR in. It appears that @infotroph has submitted a PR to @juliusvira branch of pecan with some fixes that should resolve the GH Actions warnings. Could you pull that in? That should update this PR and then we should be good to go!

**Pull request merged**